### PR TITLE
feat: Resurrected the STATIC flag reason. Documented the caching strategy.

### LINKED ISSUE #223: Define and document a caching strategy

### PR DESCRIPTION
**Pull request by skyerus**
## This PR

Resurrected the STATIC flag reason. Documented the caching strategy.

### Related Issues

Fixes #223

### CODE REVIEW

File: docs/caching.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+```
+if reason == "STATIC" {
+    isFlagCacheable = true
+}
+```
+
+
+
+#### Busting
```
Comment (toddbaert) [inline docs/caching.md:19]:
```suggestion
#### Cache invalidation
```

Comment (toddbaert) [reply docs/caching.md:19]: I won't comment on possible misinterpretations of the previous heading.

**Comment by toddbaert** [comment]
This makes sense to me. I think this should enable a simple but effective caching strategy on the clients.

**Review by toddbaert — APPROVED** [review]

**Review by toddbaert — APPROVED** [review]

File: docs/caching.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+### Caching
+
+`flagd` has a caching strategy implementable by providers that support server-to-client streaming.
+
+#### Cacheable flags
+
+`flagd` sets the `reason` of a flag evaluation as `STATIC` when no targeting rules are configured for the flag. A client can safely store the result of a static evaluation in its cache indefinitely (until the configuration of the flag changes, see [busting](#busting)).
```
Comment (skyerus) [inline docs/caching.md:7]:
```suggestion
`flagd` sets the `reason` of a flag evaluation as `STATIC` when no targeting rules are configured for the flag. A client can safely store the result of a static evaluation in its cache indefinitely (until the configuration of the flag changes, see [cache invalidation](#cache-invalidation)).
```

**Review by james-milligan — APPROVED** [review]

**Review by AlexsJones — APPROVED** [review]

**Pull request merged**